tests/test_metadata_tools.py

```diff
@@ -2,26 +2,76 @@
 import pandas as pd
 import shutil
 from metadata_tools import MetadataTools
+from EXIF_constants import EXIFConstants
+import xml.etree.ElementTree as ET
+import subprocess
+
 class TestMetadataTools(unittest.TestCase):
 
+    @classmethod
+    def get_exif_tags(self):
+        """Helper method to extract complete list of exif terms compatible with exiftool"""
+        result = subprocess.run(['exiftool', '-listx', '-f'], capture_output=True, text=True)
+
+        if result.returncode != 0:
+            raise Exception(f"Error running exiftool: {result.stderr}")
+
+        xml_output = result.stdout
+
+        #XML output --> string
+        root = ET.fromstring(xml_output)
+
+        tags = set()
+        for table in root.findall('table'):
+            for tag in table.findall('tag'):
+                tag_name = tag.get('name')
+                tags.add(tag_name)
+
+            # Check for IFD0 group
+            if table.get('name') == 'IFD0':
+                for tag in table.findall('tag'):
+                    tag_name = f"IFD0:{tag.get('name')}"
+                    tags.add(tag_name)
+
+        return tags
+
+
     def setUp(self):
         self.path = "tests/test_images/test_image.jpg"
         self.md = MetadataTools(path=self.path)
+        self.valid_exif_tags = self.get_exif_tags()
         shutil.copyfile("tests/test_images/test_image.jpg", "tests/test_images/image_backup.jpg")
 
+
     def test_read_exif_tags(self):
         """tests exif read function"""
         exif_dict = self.md.read_exif_tags()
         self.assertFalse(pd.isna(exif_dict))
         self.assertNotEqual(exif_dict, {})
-        self.assertEqual(exif_dict[272], 'iPhone XR')
+        self.assertEqual(exif_dict['EXIF:LensMake'], 'Apple')
 
     def test_write_exif_tags(self):
-        pass
+        """tests exif attach function"""
+        exif_dict = {'EXIF:LensMake': 'Samsung', 'IPTC:City': 'San Boston', 'EXIF:ApertureValue': '1.5'}
+        self.md.write_exif_tags(exif_dict=exif_dict)
+        exif_return = self.md.read_exif_tags()
+        self.assertEqual("Samsung", exif_return['EXIF:LensMake'])
+        self.assertEqual('San Boston', exif_return['IPTC:City'])
+        self.assertEqual('1.5', exif_return['EXIF:ApertureValue'])
 
```

Comment:
tests whether all exif tags in EXIF constants base class are acceptable to exiftool.